src/components/TextField.vue

```diff
@@ -456,16 +489,31 @@ export default {
       &::-webkit-scrollbar-thumb {
         @apply w-1 bg-gray-300 rounded;
       }
+
       @apply resize-none overflow-auto h-20;
     }
   }
 
+  &--currency {
+    .text-field__currency {
+      @apply block;
+    }
+
+    .text-field__input {
+      @apply pl-6;
+    }
+  }
+
   &:not(.text-field--disabled) .text-field__input {
     &:focus,
     &:active {
       + .text-field__label {
         @extend %filled-transformation;
       }
+
+      ~ .text-field__currency {
```

Comment:
Do you need this "direct descendant" selector (or whatever it's called)?

Reply:
Yes, we need it in this case.